### PR TITLE
12. Bombs Away in Python

### PR DESCRIPTION
**Pull request by bernardcooke53**
Didn't see anything pure functional and haven't done much FP myself either so figured I'd give it a go. Tested with Python 3.8.10 on Ubuntu, formatted with Black & mypy type-checked - figured that was good practice to pay forwards too. Keen to hear what you think!

### CODE REVIEW

File: 12_Bombs_Away/python/bombs_away.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+def commence_non_kamikazi_attack() -> None:
+    while True:
+        try:
+            nmissions = int(input("HOW MANY MISSIONS HAVE YOU FLOWN? "))
+
+            while nmissions >= 160:
+                print("MISSIONS, NOT MILES...")
+                print("150 MISSIONS IS HIGH EVEN FOR OLD-TIMERS")
+                nmissions = int(input("NOW THEN, HOW MANY MISSIONS HAVE YOU FLOWN? "))
+                break
```
Comment (johnflavin) [inline 12_Bombs_Away/python/bombs_away.py:54]: With this `break` where it is now it looks like this inner `while` loop only runs once.  Did you mean for this `break` to be dedented one level?

**Comment by johnflavin** [comment]
This got merged as-is, so I'll propose my changes as a follow-up MR. Probably more efficient that way anyway. 😄 

Thanks @bernardcooke53!

**Comment by coding-horror** [comment]
Sure, PRs to improve things are always welcome @johnflavin !

**Comment by bernardcooke53** [comment]
Thanks to you both @johnflavin and @coding-horror!

File: 12_Bombs_Away/python/bombs_away.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+Bombs away
+
+Ported from BASIC to Python3 by Bernard Cooke (bernardcooke53)
+Tested with Python 3.8.10, formatted with Black and type checked with mypy.
+"""
+import random
+from typing import Iterable
+
+
+def _stdin_choice(prompt: str, *, choices: Iterable[str]) -> str:
```
[the remark this answers is not in the document]
Comment (bernardcooke53) [reply 12_Bombs_Away/python/bombs_away.py:11]: FWIW - got held up today, I meant to come back and add this in too. Completely agree that this is cleaner & TIL `typing.Container`. I work with 3.6+ compatibility by day so completely forgot about the walrus operator - thank you for the guidance.

**Pull request merged**